skymp5-server/cpp/unit/DistContentsTest.h

```diff
@@ -0,0 +1,87 @@
+#include "TestUtils.hpp"
+#include <catch2/catch.hpp>
+
+namespace {
+size_t PrintMissing(const std::set<std::filesystem::path>& whatIsMissing,
+                    const std::set<std::filesystem::path>& whereIsMissing,
+                    char sign, std::ostream& out)
+{
+  std::vector<std::filesystem::path> missing;
+  for (auto& p : whatIsMissing) {
+    if (whereIsMissing.count(p) == 0) {
+      missing.push_back(p);
+    }
+  }
+  if (!missing.empty()) {
+    for (auto& p : missing) {
+      out << ' ' << sign << ' ' << p << "\n";
+    }
+  }
+  return missing.size();
+}
+
+auto GetExpectedPaths(const nlohmann::json& j)
+{
+  std::set<std::filesystem::path> res;
+
+#ifdef NDEBUG
+  constexpr auto currentConfig = "Release";
+#else
+  constexpr auto currentConfig = "Debug";
+#endif
+
+  for (auto& entry : j) {
+    std::string path;
+    std::string config;
+    if (entry.is_string()) {
+      path = entry.get<std::string>();
+    } else {
+      path = entry["path"];
+      config =
+        entry.contains("config") ? entry["config"].get<std::string>() : "";
+    }
+    if (config.empty() || config == currentConfig) {
+      res.insert(path);
+    }
+  }
+
+  return res;
+}
+}
+
+TEST_CASE("Distribution folder must contain all requested files",
+          "[DistContents]")
+{
+  auto distDir = std::filesystem::u8path(DIST_DIR);
+  auto begin = std::filesystem::recursive_directory_iterator(distDir);
+  auto end = std::filesystem::recursive_directory_iterator();
+
+  std::set<std::filesystem::path> paths;
+
+  for (auto it = begin; it != end; ++it) {
+    if (!it->is_directory()) {
+      auto& path = it->path();
+      paths.insert(std::filesystem::relative(path, distDir));
+    }
+  }
+
+  nlohmann::json j;
+  std::ifstream f(std::filesystem::u8path(__FILE__).parent_path() /
+                  "DistContentsExpected.json");
+  f >> j;
+
+  const std::set<std::filesystem::path> expectedPaths = GetExpectedPaths(j);
+
+  size_t totalMissing = 0;
+  std::stringstream ss;
+  totalMissing += PrintMissing(paths, expectedPaths, '+', ss);
+  totalMissing += PrintMissing(expectedPaths, paths, '-', ss);
```

Comment:
This is probably quite arguable, but I think `+` and `-` should be swapped. Why? From my point of view:
* `+` means like 'this result is correct, because it is in correct files list, however it's missing from actual list for some reason'
* `-` means 'this should be removed from dist, because it's missing from correct files list

Also, we could print a legend describing what `-` and `+` actually mean next to diff.

Reply:
Printing a legend makes sense, but I prefer diff-like semantics

Reply:
I agree about diff-like sepantics, but it may be unclear which `-` and `+` mean at first look. However it can be easily tested by manually checking dist dir